performance/autoupdate-clients/main.go

```diff
@@ -0,0 +1,258 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"log"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"os"
+	"os/signal"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+)
+
+func main() {
+	N := getNFromCommandLine()
+	host, user, pw := getParams()
+	jar := getCookieJar(user, pw, host)
+
+	sumBytes := 0
+	oldSumKB := 0
+	changeIds := make(map[int]int)
+	changeIdsMutex := &sync.Mutex{}
+	gotUpdate := false
+
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt)
+
+	quit := make(chan int)
+	recv := make(chan Message)
+	for i := 0; i < N; i++ {
+		go client(quit, recv, i, jar, host)
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	go func(quit chan int) {
+		for {
+			select {
+			case <-quit:
+				return
+			case <-time.After(time.Second):
+				if !gotUpdate {
+					continue
+				}
+				gotUpdate = false
+
+				// count the amount of change ids per change id
+				amountChangeIds := make(map[int]int)
+				changeIdsMutex.Lock()
+				for _, changeId := range changeIds {
+					if _, ok := amountChangeIds[changeId]; ok {
+						amountChangeIds[changeId] += 1
+					} else {
+						amountChangeIds[changeId] = 1
```

Comment:
This block is not necessary. Each key in a map has the 0-value as default. If you write
```go
amountChangeIDs[changeId] += 1
```
It does the same. Each map in go is like a defaultdict in python